arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARRoutePlannerViewController.swift

```diff
@@ -0,0 +1,240 @@
+    /// The data source to track device location and provide updates to location display.
+    let locationDataSource = AGSCLLocationDataSource()
+    /// A graphic overlay for start and destination graphics.
+    let stopGraphicsOverlay = AGSGraphicsOverlay()
+    
+    /// A graphic overlay for route graphics.
+    let routeGraphicsOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+        overlay.renderer = AGSSimpleRenderer(
+            symbol: AGSSimpleLineSymbol(style: .solid, color: .yellow, width: 5)
+        )
+        return overlay
+    }()
+    
+    /// An OAuth2 configuration to access online routing service.
+    let oAuthConfiguration = AGSOAuthConfiguration(
+        portalURL: URL(string: "https://www.arcgis.com")!,
+        clientID: "lgAdHkYZYlwwfAhC",
+        redirectURL: "my-ags-app://auth"
+    )
+    
+    /// An `AGSPoint` representing the start of navigation.
+    var startPoint: AGSPoint? {
+        didSet {
+            resetButtonItem.isEnabled = true
+            let stopSymbol = AGSPictureMarkerSymbol(image: UIImage(named: "StopA")!)
+            let startStopGraphic = AGSGraphic(geometry: self.startPoint, symbol: stopSymbol)
+            stopGraphicsOverlay.graphics.add(startStopGraphic)
+        }
+    }
+    
+    /// An `AGSPoint` representing the destination of navigation.
+    var endPoint: AGSPoint? {
+        didSet {
+            let stopSymbol = AGSPictureMarkerSymbol(image: UIImage(named: "StopB")!)
+            let endStopGraphic = AGSGraphic(geometry: self.endPoint, symbol: stopSymbol)
+            stopGraphicsOverlay.graphics.add(endStopGraphic)
+        }
+    }
+    
+    /// The route result solved by the route task.
+    var routeResult: AGSRouteResult? {
+        willSet(newValue) {
+            // Only enable when there is a valid route to navigate.
+            if newValue?.routes.first != nil {
+                navigateButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    // MARK: Instance methods
+    
+    /// Create the start and destination stops for the navigation.
+    ///
+    /// - Returns: An array of `AGSStop` objects.
+    func makeStops() -> [AGSStop] {
+        let stop1 = AGSStop(point: self.startPoint!)
+        stop1.name = "Start"
+        let stop2 = AGSStop(point: self.endPoint!)
+        stop2.name = "Destination"
+        return [stop1, stop2]
+    }
+    
+    /// A wrapper function for operations after the route is solved by an `AGSRouteTask`.
+    ///
+    /// - Parameter routeResult: The result from `AGSRouteTask.solveRoute(with:completion:)`.
+    func didSolveRoute(with routeResult: Result<AGSRouteResult, Error>) {
+        switch routeResult {
+        case .success(let routeResult):
+            self.routeResult = routeResult
+            if let firstRoute = routeResult.routes.first!.routeGeometry {
+                let routeGraphic = AGSGraphic(geometry: firstRoute, symbol: nil)
```

Comment:
```suggestion
            if let firstRoute = routeResult.routes.first {
                let routeGraphic = AGSGraphic(geometry: firstRoute.routeGeometry, symbol: nil)
```
Ha I pasted to wrong place. Will fix this.